bigquery/google/cloud/bigquery/job.py

```diff
@@ -1949,6 +1953,29 @@ def result(self, timeout=None, retry=DEFAULT_RETRY):
         return self._client.list_rows(dest_table, selected_fields=schema,
                                       retry=retry)
 
+    def to_dataframe(self):
+        """Create a pandas DataFrame from the query results.
```

Comment:
What's the reason this is in the QueryJob and not a subclass of the paged iterator we use to list rows?

I think this method could be handy to use on the result of `client.list_rows(table)`.

Reply:
The Iterator class is in api_core and we only want to call `to_dataframe()` on the iterators of query results, so I thought it would make more sense to have it here. Would it make sense to have a special type of iterator for query results?

Reply:
Per https://github.com/GoogleCloudPlatform/google-cloud-python/issues/4152#issuecomment-336952584 @dhermes recommends we create a custom iterator class to address that issue, anyway. I'd prefer we do it now and add the `to_dataframe()` to that class.

Note: there isn't anything special about the query results, if you trace through the code you will see that it calls `list_rows` behind the scenes.

Reply:
(I'm not saying fix #4152 in this PR, just add the custom class)